laurent22/joplin
# Chore: Refactor mobile plugin logic into locations more consistent with other parts of the app

# Summary

This pull request refactors backend logic for mobile plugins into directories more consistent with Joplin's existing codebase. In particular,
- `loadPlugins.ts` and some of the plugin testing logic is moved to `lib/services/plugins/`.
- The mobile `PlatformImplementation` is moved into `app-mobile/services/plugins/`
- Other mobile plugin logic (e.g. the plugin runner component) is moved into `app-mobile/components/`

# Testing plan

1. Start Joplin.
2. Enable plugins.
3. Uninstall all installed plugins (if any).
4. Install "Note tabs".
5. Wait several seconds and verify that the "Plugin panels" button is visible in the toolbar.
6. Disable "Note tabs".
7. Verify that the plugin panel toggle is no longer in the toolbar.
8. Enable "Note tabs".
9. Verify that the plugin panel is present.
10. Install "Rich Markdown" and enable "Render images below their markdown source".
11. Edit a note and add a drawing.
12. Verify that Rich Markdown renders the drawing below its markdown source.
13. Uninstall both plugins.
14. Verify that the plugin panel is hidden.

This has been tested successfully on Android 13.
